### PR TITLE
Implement modulo for duration

### PR DESCRIPTION
**Pull request by gaetschwartz**
# Description

This PR adds the ability to use modulo with durations:

```nu
(2min + 31sec) mod 20sec # 11sec
```

# User-Facing Changes

Allows to use `<duration> mod <duration>`

### CODE REVIEW

**Comment by fdncred** [comment]
Interesting. I'm not sure I've ever had the use case to do mod on durations.

**Comment by gaetschwartz** [comment]
Durations manipulations as a whole is pretty rare for me as well, just ended up needing to use modulo while writing a benchmark 😄

**Review by amtoine — DISMISSED** [review]
sounds legit to me, because `duration`s are really really close to number, just with a unit :ok_hand: 

@gaetschwartz 
could you add some tests to cover this new behaviour? :blush:

**Comment by gaetschwartz** [comment]
I was planning to but was not sure where it should be added, I can't seem to find tests for modulo-ing other values ?

**Comment by fdncred** [comment]
since `mod` is an operator, maybe you can find where other operators are tested and throw it in there. i assume there's some operator tests somewhere.

File: crates/nu-protocol/src/value/mod.rs [hunk 1]
```diff
@@ -3445,6 +3445,13 @@ impl Value {
                     Err(ShellError::DivisionByZero { span: op })
                 }
             }
+            (Value::Duration { val: lhs, .. }, Value::Duration { val: rhs, .. }) => {
+                if *rhs != 0 {
+                    Ok(Value::duration(lhs % rhs, span))
```
Comment (bobhy) [inline crates/nu-protocol/src/value/mod.rs:3450]: Should be `Value::Int()`, not duration.

The time units cancel out in this division-like operation.

See `src/tests/test_math.rs` for a place to implement some unit tests.

**Comment by bobhy** [comment]
This is a useful operation!  e.g
`365day mod 1wk` -> `1day`
Tells you the day of the week one year from today, a Tuesday, will be a Wednesday.

**Review by sholderbach — APPROVED** [review]
Thanks for the improvement and the tests!

**Pull request merged**